core/domain/stats_domain.py

```diff
@@ -587,31 +583,35 @@ def __init__(self, calculation_output_type):
 
 class AnswerFrequencyList(AnswerCalculationOutput):
     """Domain object that represents a list of AnswerOccurrences."""
-    def __init__(self, answer_occurrences=None, limit=None):
+    def __init__(self, answer_occurrences):
         """Initialize domain object for answer frequency list for a given list
         of AnswerOccurrence objects (default is empty list).
         """
         super(AnswerFrequencyList, self).__init__(
             CALC_OUTPUT_TYPE_ANSWER_FREQUENCY_LIST)
-        answer_occurrences = answer_occurrences or []
-        self._answer_counter = utils.OrderedCounter(
-            answer_occurrence.to_pair()
-            for answer_occurrence in answer_occurrences)
-        self.limit = limit
-
-    def add_answer(self, answer):
+        self.answer_counter = utils.OrderedCounter()
+        for answer_occurrence in answer_occurrences:
+            self.answer_counter.update({
+                answer_occurrence.hashable_answer: answer_occurrence.frequency
+            })
+        self._len_limit = None
+
+    def add_answer(self, hashable_answer):
         """Adds a new Answer object."""
-        self._answer_counter[answer] += 1
+        self.answer_counter[hashable_answer] += 1
 
-    def add_answers(self, answers):
+    def add_answers(self, hashable_answers):
         """Adds an iterable of new Answer objects."""
-        self._answer_counter.update(answers)
+        self.answer_counter.update(hashable_answers)
+
+    def truncate(self, len_limit):
```

Comment:
Just for clarity: when I suggested truncate(), I envisioned that being a permanent truncation operation on the list, i.e. truncate(2) would permanently modify [a, b, c] to [a, b]. If we are looking to support queries-with-limits instead then a better name for this might be get_top_answers(limit) or similar.